book/src/implemented-proposals/repair-service.md

```diff
@@ -2,48 +2,115 @@
 
 ## Repair Service
 
-The RepairService is in charge of retrieving missing shreds that failed to be delivered by primary communication protocols like Avalanche. It is in charge of managing the protocols described below in the `Repair Protocols` section below.
+The RepairService is in charge of retrieving missing shreds that failed to be
+delivered by primary communication protocols like Turbine. It is in charge of
+managing the protocols described below in the `Repair Protocols` section below.
 
 ## Challenges:
 
 1\) Validators can fail to receive particular shreds due to network failures
 
-2\) Consider a scenario where blockstore contains the set of slots {1, 3, 5}. Then Blockstore receives shreds for some slot 7, where for each of the shreds b, b.parent == 6, so then the parent-child relation 6 -&gt; 7 is stored in blockstore. However, there is no way to chain these slots to any of the existing banks in Blockstore, and thus the `Shred Repair` protocol will not repair these slots. If these slots happen to be part of the main chain, this will halt replay progress on this node.
-
-3\) Validators that find themselves behind the cluster by an entire epoch struggle/fail to catch up because they do not have a leader schedule for future epochs. If nodes were to blindly accept repair shreds in these future epochs, this exposes nodes to spam.
-
-## Repair Protocols
-
-The repair protocol makes best attempts to progress the forking structure of Blockstore.
+2\) Consider a scenario where blockstore contains the set of slots {1, 3, 5}.
+Then Blockstore receives shreds for some slot 7, where for each of the shreds
+b, b.parent == 6, so then the parent-child relation 6 -&gt; 7 is stored in 
+blockstore. However, there is no way to chain these slots to any of the 
+existing banks in Blockstore, and thus the `Shred Repair` protocol will not
+repair these slots. If these slots happen to be part of the main chain, this
+will halt replay progress on this node.
+
+## Repair-related primitives
+Epoch Slots: 
+   Each validator advertises separately on gossip thhe various parts of an 
+   `Epoch Slots`:
+   * The `stash`: An epoch-long compressed set of all completed slots.
```

Comment:
Is this going to fit in an MTU?

Reply:
Stash can be split amongst N entries in the table.